vowpalwabbit/slim/include/vw_slim_return_codes.h

```diff
@@ -0,0 +1,16 @@
+#pragma once
+
+#define S_VW_PREDICT_OK									0
+#define E_VW_PREDICT_ERR_INVALID_MODEL					1
+#define E_VW_PREDICT_ERR_WEIGHT_INDEX_OUT_OF_RANGE		2
+#define E_VW_PREDICT_ERR_GD_RESUME_NOT_SUPPORTED		3
+#define E_VW_PREDICT_ERR_CB_EXPLORATION_MISSING			4
+#define E_VW_PREDICT_ERR_NOT_A_CB_MODEL					5
+#define E_VW_PREDICT_ERR_NO_MODEL_LOADED				6
+#define E_VW_PREDICT_ERR_NO_A_CSOAA_MODEL				7
+#define E_VW_PREDICT_ERR_EXPLORATION_FAILED				8
+#define E_VW_PREDICT_ERR_INVALID_MODEL_CHECK_SUM		9
+#define E_VW_PREDICT_ERR_HASH_SEED_NOT_SUPPORTED		10
+#define COMMA ,
```

Comment:
Why?

Reply:
Macro expansion does not like commas in some cases.

Reply:
Macro expansion understands nested parentheses. I often see simply the parameter wrapped in brackets

Reply:
The issue is usage of template args in macros.  This is a known issue in C++.

For example the following:
[https://github.com/rajan-chari/vowpal_wabbit/blob/9eb9aa7371e3cbed2010c889e80d8928038e5a80/vowpalwabbit/slim/include/model_parser.h#L42](https://github.com/rajan-chari/vowpal_wabbit/blob/9eb9aa7371e3cbed2010c889e80d8928038e5a80/vowpalwabbit/slim/include/model_parser.h#L42)


---
In reply to: [304041484](https://github.com/VowpalWabbit/vowpal_wabbit/pull/1800#discussion_r304041484) [](ancestors = 304041484)

Reply:
I understand, but wrapping in brackets resolves this.

Reply:
Not sure what you mean...

This works:
RETURN_ON_FAIL(mp.read<uint32_t COMMA false>("check_sum_len", check_sum_len));

This does not compile:
    RETURN_ON_FAIL({mp.read<uint32_t , false>("check_sum_len", check_sum_len)});

---
In reply to: [304452335](https://github.com/VowpalWabbit/vowpal_wabbit/pull/1800#discussion_r304452335) [](ancestors = 304452335)